rpc/core/net.go

```diff
@@ -13,20 +13,19 @@ import (
 // NetInfo returns network info.
 // More: https://docs.cometbft.com/main/rpc/#/Info/net_info
 func (env *Environment) NetInfo(*rpctypes.Context) (*ctypes.ResultNetInfo, error) {
-	peersList := env.P2PPeers.Peers().List()
-	peers := make([]ctypes.Peer, 0, len(peersList))
-	for _, peer := range peersList {
-		nodeInfo, ok := peer.NodeInfo().(p2p.DefaultNodeInfo)
+	peers := make([]ctypes.Peer, 0)
+	env.P2PPeers.Peers().ForEach(func(p p2p.Peer) {
+		nodeInfo, ok := p.NodeInfo().(p2p.DefaultNodeInfo)
 		if !ok {
-			return nil, fmt.Errorf("peer.NodeInfo() is not DefaultNodeInfo")
+			panic(fmt.Sprintf("peer %v has the invalid node info type: %T ", p.ID(), p.NodeInfo()))
```

Comment:
note that we refuse handshake if `p.NodeInfo()` is not `DefaultNodeInfo`, but I can revert the code to error instead.

Reply:
But let's play safe.